shnet_main.c

```diff
@@ -190,21 +192,22 @@ static int shnet_port_recv(struct shnet_port *port,
     recv->vec = recv->req.vec;
     recv->port = port;
 
-    if ((unsigned long)recv->req.vec[0].iov_base & (PAGE_SIZE -1)) {
+    nr_pages = recv->req.vec[0].iov_len / PAGE_SIZE + 1;
+    pg_offs = (unsigned long)recv->req.vec[0].iov_base & (PAGE_SIZE - 1);
+    if (pg_offs) {
         pr_err("Address %p is not aligned\n", recv->req.vec[0].iov_base);
-    	return post_cqe(port, recv->req.connection_id, recv->req.req_id,
-			SHNET_ERR_CODE_INV_ADDR);
+	nr_pages++;
```

Comment:
No because it is not taking into account that fact that i had to move to page boundary address, please recall our example of the address 0x1010.
When use asked for one page starting address  0x1010 the first calculation will gives us 1 but because the address is not aligned and i have to lock from 0x1000 i now need to lock 2 pages because 1 will lock only 0x1000 to 0x2000 while user requested to 0x2010.